src/ui/units/collections/components/CollectionContent/CollectionContent.tsx

```diff
@@ -245,235 +327,164 @@ export const CollectionContent = React.memo<Props>(
-                    action: () => {
-                        setDialogState(DialogState.EditWorkbookAccess);
-                        setDialogEntity(item);
-                    },
-                });
-            }
-            const otherActions: DropdownMenuItem[] = [];
-            if (item.permissions.delete) {
-                otherActions.push({
-                    text: <DropdownAction icon={TrashBin} text={i18n('action_delete')} />,
-                    action: () => {
-                        setDialogState(DialogState.DeleteWorkbook);
-                        setDialogEntity(item);
-                    },
-                    theme: 'danger',
-                });
-            }
+        actions.push([...otherActions]);
+        return actions;
+    };
 
-            actions.push([...otherActions]);
-            return actions;
-        };
+    const contentProps = {
+        contentItems,
+        filters,
+        setFilters,
+        getWorkbookActions,
+        getCollectionActions,
+        onUpdateCheckbox,
+        onSelectAll,
+        selectedMap,
+        countSelected,
+    };
 
-        return (
-            <React.Fragment>
-                {collectionPageViewMode === CollectionPageViewMode.Grid ? (
-                    <CollectionContentGrid
-                        contentItems={contentItems}
-                        filters={filters}
-                        setFilters={setFilters}
-                        getWorkbookActions={getWorkbookActions}
-                        getCollectionActions={getCollectionActions}
-                    />
-                ) : (
-                    <CollectionContentTable
-                        contentItems={contentItems}
-                        filters={filters}
-                        setFilters={setFilters}
-                        getWorkbookActions={getWorkbookActions}
-                        getCollectionActions={getCollectionActions}
-                    />
-                )}
-
-                {isContentLoading && <SmartLoader className={b('loader')} size="m" showAfter={0} />}
-                {!isContentLoading && !waypointDisabled && <Waypoint onEnter={onWaypointEnter} />}
-                {dialogEntity && (
-                    <React.Fragment>
-                        {'workbookId' in dialogEntity ? (
-                            <React.Fragment>
-                                <DeleteWorkbookDialogContainer
-                                    open={dialogState === DialogState.DeleteWorkbook}
-                                    workbookId={dialogEntity.workbookId}
-                                    workbookTitle={dialogEntity.title}
-                                    deleteInItems={true}
+    return (
+        <React.Fragment>
+            {collectionPageViewMode === CollectionPageViewMode.Grid ? (
+                <CollectionContentGrid
+                    {...contentProps}
```

Comment:
Using object destructuring when propagating props is not completely type safe. We can add any additional props and typescript won't highlight them.